src/UI/templates/js/MainControls/mainbar.js

```diff
@@ -0,0 +1,213 @@
+il = il || {};
+il.UI = il.UI || {};
+il.UI.maincontrols = il.UI.maincontrols || {};
+
+(function($, maincontrols) {
+	maincontrols.mainbar = (function($) {
+
+		var id
+			,_cls_btn_engaged = 'engaged'
+			,_cls_page_active_slates = 'with-mainbar-slates-engaged' //set on _page_div
+			,_cls_entries_wrapper = 'il-mainbar-entries' //encapsulating div of all entries
+			,_cls_toolentries_wrapper = 'il-mainbar-tools-entries' //tools (within  _cls_entries_wrapper)
```

Comment:
I wonder if we should have some naming rules that distinguish style classes that are referenced in JS from style classes that are for design purposes only. Somthing that tells the person that modifies a template: Changing this name will break the behaviour.

Reply:
Sounds very plausible and would also be applicable in other contexts (looking at you, filter), right? This would imo also require that a class is either for styling or for behaviour, not both. Should we take this discussion elsewhere, maybe into a proposal for the roadmap of the UI-framework? Does someone feel entitle to write a proposal?